### PR TITLE
TE-441 general-widgets unit test refactor

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-441)

### What **one** thing does this PR do?
consume @lodgify/enzyme-jest-expect-helpers in the unit tests

### CODE REVIEW

File: src/components/general-widgets/HTML/component.spec.js [hunk 1]
```diff
@@ -1,69 +1,56 @@
...
 import { Component as HTMLWidget } from './component';
 
 const { headings } = require('./mock-data/examples');
 
 describe('<HTML />', () => {
   it('should render a single Lodgify UI `HTML` component', () => {
-    const htmlWidget = shallow(<HTMLWidget />);
-
-    expect(htmlWidget.find('div').length).toBeGreaterThan(0);
+    const wrapper = shallow(<HTMLWidget />);
```
Comment (Jonnygwi) [inline src/components/general-widgets/HTML/component.spec.js]: `shallow(<HTMLWidget />)` is used in a lot of unit tests. Turn this into a reusable function.
`const getHTMLWidget = () => shallow(<HTMLWidget />);`

File: src/components/general-widgets/HTML/component.spec.js [hunk 2]
```diff
@@ -1,69 +1,56 @@
...
+    expect(wrapper.find('div').length).toBeGreaterThan(0);
```
Comment (Jonnygwi) [inline src/components/general-widgets/HTML/component.spec.js]: use the `expectComponentToBe` helper here. `expectComponentToBe(wrapper, 'div')`

File: src/components/general-widgets/HTML/component.spec.js [hunk 3]
```diff
@@ -1,69 +1,56 @@
...
   });
 
   it('should contain a HTML component with the right props', () => {
-    const htmlWidget = shallow(<HTMLWidget htmlString={headings} />).find(
-      'div'
-    );
-
-    expect(htmlWidget).toHaveLength(1);
+    const wrapper = shallow(<HTMLWidget htmlString={headings} />).find('div');
```
Comment (Jonnygwi) [inline src/components/general-widgets/HTML/component.spec.js]: update the `getHTMLWidget` function to take props and use it here
`const getHTMLWidget = (props) => shallow(<HTMLWidget {...props} />);`

File: src/components/general-widgets/HTML/component.spec.js [hunk 4]
```diff
@@ -1,69 +1,56 @@
...
-
-      const firstDiv = htmlWidget.find('div').render();
+      const wrapper = mount(<HTMLWidget htmlString={headings} />);
 
-      expect(firstDiv.find('h1')).toHaveLength(1);
-      expect(firstDiv.find('h2')).toHaveLength(0);
-      expect(firstDiv.find('h3')).toHaveLength(1);
-      expect(firstDiv.find('h4')).toHaveLength(1);
-      expect(firstDiv.find('h5')).toHaveLength(1);
+      expectComponentToHaveChildren(wrapper, 'div');
```
Comment (Jonnygwi) [inline src/components/general-widgets/HTML/component.spec.js]: Is this testing the right thing?

Comment (red-runner) [reply src/components/general-widgets/HTML/component.spec.js]: @Jonnygwi I had the same concern about this particular case. I checked it up with @marcdavi-es and he concluded that it's doing what it should 👌

File: src/components/general-widgets/HTML/component.spec.js (hunk 3 above)
Comment (red-runner) [reply src/components/general-widgets/HTML/component.spec.js]: nice tip 🔝

File: src/components/general-widgets/HTML/component.spec.js (hunk 4 above)
Comment (marcdavi-es) [reply src/components/general-widgets/HTML/component.spec.js]: Yeah I think that's fine :)

**Pull request merged**